### ISSUE TITLE
a11y: Date picker

### ISSUE THREAD

**Issue by staceytan1998** [opengovsg/FormSG#5519]
Currently, 

- Users type out the date according to ddmmyyyy which is what the screenreader reads out to them, they do not know there is a datepicker at the end of the field

- Date picker behaves well on JAWS but NVDA is inconsistent

- On iPhone VoiceOver it reads out a lot of unnecessary stuff, when I focus on 1st December, screen reader reads 'Thursday, Thursday Dec 01 2022, column 5 toggle button not pressed, 2 of 1, double tap to toggle setting', Mac is ok

**Suggestions**

- Currently if user tries to select a date that is disabled, it says unticked, can we say 'this date cannot be selected'

- Can we read out each date in this format, 'Monday, 27 March 2023'? Right now the date picker reads out the short form for the days and months, sounds weird

- When users are on the Date field, can we announce 'Type in the date, or use date picker to choose a date", so they know there is a date picker available
